HeterogeneousCore/AlpakaCore/test/testAlpakaBackendFilter.py

```diff
@@ -0,0 +1,41 @@
+import os
+import sys
+
+# choose a different alpaka backend depending on the SCRAM test being run
+try:
+    backend = os.environ['SCRAM_TEST_NAME'].replace('testAlpakaBackendFilter', '')
+except:
+    backend = 'SerialSync'
+
+# map the alpaka backends to the process accelerators
+accelerators = {
+    'SerialSync': 'cpu',
+    'CudaAsync':  'gpu-nvidia',
+    'ROCmAsync':  'gpu-amd'
+}
+
+import FWCore.ParameterSet.Config as cms
+from HeterogeneousCore.AlpakaCore.functions import *
+
+process = cms.Process('Test')
+
+process.options.accelerators = [ accelerators[backend] ]
+
+process.maxEvents.input = 10
+
+process.source = cms.Source('EmptySource')
+
+process.load('Configuration.StandardSequences.Accelerators_cff')
+process.load('HeterogeneousCore.AlpakaCore.ProcessAcceleratorAlpaka_cfi')
+
+process.alpakaBackendProducer = cms.EDProducer('AlpakaBackendProducer@alpaka')
+
+process.alpakaBackendFilter = cms.EDFilter('AlpakaBackendFilter',
+  producer = cms.InputTag('alpakaBackendProducer', 'backend'),
+  backends = cms.vstring(backend)
+)
+
+process.SelectedBackend = cms.Path(process.alpakaBackendProducer + process.alpakaBackendFilter)
+process.AnyOtherBackend = cms.Path(process.alpakaBackendProducer + ~process.alpakaBackendFilter)
```

Comment:
> > The test doesn't seem to really test the behavior of `AlpakaBackendFilter`.
> 
> What do you mean ?

I mean the filtering decision being the expected one is not being tested. E.g. something along
```suggestion
process.mustRun = cms.EDProducer("edmtest::MustRunIntProducer", ivalue=cms.int32(1))
process.mustNotRun = cms.EDProducer("FailingProducer")

process.SelectedBackend = cms.Path(process.alpakaBackendProducer + process.alpakaBackendFilter + process.mustRun)
process.AnyOtherBackend = cms.Path(process.alpakaBackendProducer + ~process.alpakaBackendFilter + process.mustNotRun)
```
would lead to the job to fail if the filtering decision (i.e. the joint logic in `AlpakaBackendFilter` constructor and `filter()` function) would be unexpected for any reason.

Reply:
Ah, I see.
I guess it's enough to have the `FailingProducer` ?
Is that a real module one can use, or just an example ?

Reply:
> I guess it's enough to have the `FailingProducer` ?

Right, in this case where the "other behavior" is achieved with `~` (whose behavior we should be testing in the framework) the first case should indeed be sufficient.

If the other Path would use another instance of `AlpakaBackendFilter` with different `backend` configuration, then testing it explicitly could be useful.

> Is that a real module one can use, or just an example ?

It is available (from test modules defined in `FWCore/Framework/test`).

Reply:
OK, thanks, I'll make the changes.